### PR TITLE
vagrant: update to 2.3.0.

### PR DESCRIPTION
**Pull request by dataCobra**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

I've also updated `do_install` process.

I've added a patch to get rid of an annoying message when executing vagrant commands.

### CODE REVIEW

File: srcpkgs/vagrant/template [hunk 1]
```diff
@@ -10,8 +10,9 @@ short_desc="Tool for building and distributing development environments"
 maintainer="Andrea Brancaleoni <abc@pompel.me>"
 license="MIT"
 homepage="https://www.vagrantup.com/"
-distfiles="https://github.com/mitchellh/${pkgname}/archive/v${version}.tar.gz"
-checksum=4f0e6b1d466e26dead682c4d4843e8f64a012eba4be91506ae6c6d34d3d9c8f9
+changelog="https://github.com/hashicorp/${pkgname}/blob/v${version}/CHANGELOG.md"
+distfiles="https://github.com/hashicorp/${pkgname}/archive/v${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:14]: expand pkgname

File: srcpkgs/vagrant/template [hunk 2]
```diff
@@ -25,6 +26,16 @@ do_build() {
 do_install() {
 	GEM_PATH=$DESTDIR/usr/lib/$pkgname GEM_HOME="$GEM_PATH" gem install $pkgname-$version.gem --no-document -- --disable-clean
 
+	vlicense LICENSE
+
+	vmkdir usr/bin
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:31]: not needed for vbin

File: srcpkgs/vagrant/template [hunk 3]
```diff
@@ -25,6 +26,16 @@ do_build() {
...
+	vbin $FILESDIR/$pkgname
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:32]: expand pkgname

File: srcpkgs/vagrant/template [hunk 4]
```diff
@@ -25,6 +26,16 @@ do_build() {
...
+
+	vcompletion "contrib/bash/completion.sh" bash
+
+	mkdir -p /usr/share/vim/vim90/ftplugin/
+	cp contrib/vim/vagrantfile.vim /usr/share/vim/vim90/ftplugin/
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:37]: use vcopy or vinstall, (vinstall should get rid of the mkdir too, which is broken anyways... you'll probably want to `./xbps-src zap && ./xbps-src binary-bootstrap`, you're polluting the masterdir)

File: srcpkgs/vagrant/template [hunk 5]
```diff
@@ -10,8 +10,9 @@ short_desc="Tool for building and distributing development environments"
 maintainer="Andrea Brancaleoni <abc@pompel.me>"
 license="MIT"
 homepage="https://www.vagrantup.com/"
-distfiles="https://github.com/mitchellh/${pkgname}/archive/v${version}.tar.gz"
-checksum=4f0e6b1d466e26dead682c4d4843e8f64a012eba4be91506ae6c6d34d3d9c8f9
+changelog="https://github.com/hashicorp/${pkgname}/blob/v${version}/CHANGELOG.md"
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:13]:
```suggestion
changelog="https://raw.githubusercontent.com/hashicorp/vagrant/main/CHANGELOG.md"
```

**Review by classabbyamp — COMMENTED** [review]
does this package work with the gemspec build style?

File: srcpkgs/vagrant/template (hunk 1 above)
Comment (dataCobra) [reply srcpkgs/vagrant/template:14]: Do you mean to use `vagrant` instead of the variable `${pkgname}`?

Comment (classabbyamp) [reply srcpkgs/vagrant/template:14]: yes

File: srcpkgs/vagrant/template (hunk 4 above)
Comment (dataCobra) [reply srcpkgs/vagrant/template]: Thanks for the advice. I've changed it accordingly.

**Comment by dataCobra** [comment]
> does this package work with the gemspec build style?

I've change the build_style to `gemspec` and refactored the template a little bit.

File: srcpkgs/vagrant/template [hunk 6]
```diff
@@ -1,29 +1,34 @@
...
 homepage="https://www.vagrantup.com/"
-distfiles="https://github.com/mitchellh/${pkgname}/archive/v${version}.tar.gz"
-checksum=4f0e6b1d466e26dead682c4d4843e8f64a012eba4be91506ae6c6d34d3d9c8f9
+changelog="https://raw.githubusercontent.com/hashicorp/vagrant/main/CHANGELOG.md"
+distfiles="https://github.com/hashicorp/vagrant/archive/v${version}.tar.gz"
+checksum=1931dbf29ec3c6622a649ae145fe706e5b957d7075870ce577358dd22c3d5dca
 nocross=yes
 
 post_extract() {
 	sed -i 's|, "< 3\.1"||g' $pkgname.gemspec
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:18]:
```suggestion
	vsed -i 's|, "< 3\.1"||g' vagrant.gemspec
```

File: srcpkgs/vagrant/template [hunk 7]
```diff
@@ -1,29 +1,34 @@
...
 }
 
-do_build() {
-	gem build $pkgname.gemspec
-}
-
 do_install() {
-	GEM_PATH=$DESTDIR/usr/lib/$pkgname GEM_HOME="$GEM_PATH" gem install $pkgname-$version.gem --no-document -- --disable-clean
+	GEM_PATH=$DESTDIR/usr/lib/$pkgname GEM_HOME="$GEM_PATH" \
```
Comment (classabbyamp) [inline srcpkgs/vagrant/template:22]:
```suggestion
	GEM_PATH=$DESTDIR/usr/lib/vagrant GEM_HOME="$GEM_PATH" \
```

File: srcpkgs/vagrant/template (hunk 6 above)
Comment (classabbyamp) [inline srcpkgs/vagrant/template:18]:
```suggestion
	vsed -i 's|, "< 3\.1"||g' vagrant.gemspec
```

Comment (dataCobra) [reply srcpkgs/vagrant/template]: Changed.

File: srcpkgs/vagrant/template (hunk 7 above)
Comment (dataCobra) [reply srcpkgs/vagrant/template]: Changed.

**Comment by dataCobra** [comment]
Hey @classabbyamp,

I've also created PR #38850 for VirtualBox which is related to this PR, could you also check this one, please?

**Comment by dataCobra** [comment]
Hey @classabbyamp,

@Duncaen already merged the VirtualBox changes. Is there more work to do here?

I use this version of vagrant without any issues for 6 days now.

**Comment by classabbyamp** [comment]
cc @thypon

**Review by thypon — APPROVED** [review]

**Pull request merged**